semantics/cpp14/language/common/conversion.k

```diff
@@ -230,7 +230,25 @@ module CPP-CONVERSION
                                                 type(underlyingType(T)) 
                                                 #fi #fi #fi #fi #fi #fi
           requires T ==K char16_t orBool T ==K char32_t orBool T ==K wchar_t
-     // TODO(dwightguth): 4.5:3-5
+
+     // 4.5:3
+     // TODO(h0nzZik) (1) store the type in configuration for performance reasons,
+     //               (2) what if we have an extended integer type other then the underlying type?
+     rule promote(t(_, _, unscopedEnum(_, _, false) #as T::CPPSimpleType)) =>
+               #if broaderTypeThan(type(int), type(T)) #then type(int) #else
+               #if broaderTypeThan(type(unsigned), type(T)) #then type(unsigned) #else
+               #if broaderTypeThan(type(long), type(T)) #then type(long) #else
+               #if broaderTypeThan(type(unsigned-long), type(T)) #then type(unsigned-long) #else
+               #if broaderTypeThan(type(long-long), type(T)) #then type(long-long) #else
+               #if broaderTypeThan(type(unsigned-long-long), type(T)) #then type(unsigned-long-long) #else
+               type(underlyingType(T))
```

Comment:
This case should not actually be reachable without any extended integer types, and I would prefer a comment to that effect.